src/plugins/discover/public/services/discover_ebt_context_manager.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { BehaviorSubject } from 'rxjs';
+import { isEqual } from 'lodash';
+import type { CoreSetup } from '@kbn/core-lifecycle-browser';
+
+export interface DiscoverEBTContextProps {
+  dscProfiles: string[]; // Discover Context Awareness Profiles
+}
+export type DiscoverEBTContext = BehaviorSubject<DiscoverEBTContextProps>;
+
+export class DiscoverEBTContextManager {
+  private isEnabled: boolean = false;
+  private ebtContext$: DiscoverEBTContext | undefined;
+
+  constructor() {}
+
+  // https://docs.elastic.dev/telemetry/collection/event-based-telemetry
+  public register({ core }: { core: CoreSetup }) {
+    const context$ = new BehaviorSubject<DiscoverEBTContextProps>({
+      dscProfiles: [],
+    });
+
+    core.analytics.registerContextProvider({
+      name: 'discover_context',
+      context$,
+      schema: {
+        dscProfiles: {
+          type: 'array',
+          items: {
+            type: 'keyword',
+            _meta: {
+              description:
+                'List of profiles which are activated by Discover Context Awareness logic',
```

Comment:
```suggestion
                'List of active Discover context awareness profiles',
```
Just a suggestion for a potentially simpler description.